### PR TITLE
doc: ram_report: hidden and no paths

### PR DESCRIPTION
**Pull request by greg-fer**
Added information about Hidden and No paths.
Made minor style edits to the page.

### CODE REVIEW

**Comment by greg-fer** [comment]
@kartben , can  you have a look? Thanks!

File: doc/develop/optimizations/tools.rst [hunk 1]
```diff
@@ -76,22 +77,45 @@ which will generate something similar to the output below::
...
+The RAM report lists all processing symbols with no matching mapped files in the Hidden category.
+
+This means that the file for the listed symbol was not added to the metadata file, it was empty,
+or it was undefined. The tool was unable to get the name of the function
+for the given symbol nor identify where it comes from.
+
+No paths
+--------
+
+The RAM report lists all processing symbols with relative paths in the No paths category.
```
Comment (nordicjm) [inline doc/develop/optimizations/tools.rst:92]: This applies to ROM report also

Comment (greg-fer) [reply doc/develop/optimizations/tools.rst:92]: Moved both sections up.

**Review by Rallare — APPROVED** [review]

File: doc/develop/optimizations/tools.rst [hunk 2]
```diff
@@ -12,9 +15,29 @@ The build system offers 3 targets to view and analyse RAM, ROM and stack usage
...
+  in the No paths category.
+
+  This means that the listed symbols cannot be placed in the tree structure
+  of the report at an absolute path under one specific file. The tool was able
+  to get the name of the function, but it was unable to identify where it comes from.
+
+  .. note::
+
+     You can have multiple cases of the same function, and the No paths category
+     will list the sum of these in one entry.
```
Comment (kartben) [inline doc/develop/optimizations/tools.rst:39]: This doesn't really flow well with the previous paragraph, and would need some context for the reader who likely hasn't even used the tools just yet and therefore has no idea what "Hidden" or "No paths" refers to.

Maybe starting with something like "Some of the tools mentioned in this section are organizing their output based on the physical organization of the symbols. As some symbols might be external to the project's tree structure, or might lack metadata needed to display them by name, the following top-level containers are used to group such symbols::" would help? And maybe this whole paragraphe would benefit from being a `.. note::` (if yes, the inner note ("You can have multiple...") would probably work better visually as a simple sentence)

**Review by kartben — CHANGES_REQUESTED** [review]
This is great, thanks!  Please see my comment regarding the section about hidden / no paths which I don't think quite works as-is

File: doc/develop/optimizations/tools.rst (hunk 2 above)
Comment (greg-fer) [reply doc/develop/optimizations/tools.rst]: Good idea, @kartben . I've made the edits, but with the liberty of putting the descriptions of Hidden and No paths into a table.

**Comment by greg-fer** [comment]
@kartben , any chance we could pull this in? :)

**Review by nordicjm — DISMISSED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by greg-fer** [comment]
@kartben , edits applied, can you check again when you have some time? Thanks!

**Review by kartben — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**